environment_variable.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2016 Rohith All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package marathon
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// EnvironmentVariable is an environment variable associated with an application
+type EnvironmentVariable struct {
```

Comment:
I'm wondering if we should strive for a clearer separation between regular environment variables and secrets.

AFAICS, it should be possible to maintain separate types: `map[string]string` for environment variables and `map[string]EnvSecret` for secrets. We would probably need a custom top-level unmarshaler that deserializes the `env` JSON field into a generic `map[string]interface{}`. Iterating over that, we can do type assertions to decide if a record must go into the environment variables or secrets map, respectively.

Conversely, a custom marshaler would populate a `map[string]interface{}` from each map and let encoding/json do its serialization magic.

This approach would have the added benefit that we don't even break clients that reference  `map[string]string` types directly to work with environment variables. Plus, I think it'd turn Marathon's somewhat unconventional design decision to pack different types inside the same JSON field into a cleanly separated API.

I'm currently running short on time to tweak your PR accordingly to see if the idea could work. WDYT, should we give it a try?